python/paddle/v2/reader/decorator.py

```diff
@@ -224,3 +226,74 @@ def firstn_reader():
             yield item
 
     return firstn_reader
+
+
+class XmapEndSignal():
+    pass
+
+
+def xmap(mapper, reader, process_num, buffer_size):
+    """
+    Use multiprocess to map samples from reader by a mapper defined by user.
```

Comment:
1.  **The `buffered` decorator  is not a thread-safe data provider.**
 So i can't  put a multi-thread map decorator on top of the `buffered` decorator directly. 
2. To guarantee that handle workers read data from reader safely, multi-threads map decorator hold a queue whihin it, with which there is no need to  use bufferd decorator.